lib/google_maps.ex

```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
     GoogleMaps.get("place/queryautocomplete", params)
   end
 
+@doc """
+  Search for nearby places based on location and keywords
+
+  `nearby_search/3/4` - Most common use for Places API.
+
+  `nearby_search/1/2` - Intended for use with `rankby` 
```

Comment:
After reading Google's documents, I think we can simply have the function with arity 3, the `coordinate`, `radius`, and the `options`. The `coordinate` and `radius` are required.

Inside the function, we can check for `rankby=distance` option, and ignore the `radius`.

We can get fancy and check for existence of other options when `rankby=distance`, but as a start, we can just assume that the user will check the docs for what options are needed.

Reply:
I struggled a little with how best to approach this. Google said `radius` is required except when `rankby=distance`, which I think is kinda odd on their part. 

Going with arity of 3 and checking for `rankby=distance` makes sense.